agents-common/src/main/java/org/apache/ranger/authorization/hadoop/config/RangerConfiguration.java

```diff
@@ -17,88 +17,79 @@
-		boolean ret = false;
-
-		if(LOG.isDebugEnabled()) {
-			LOG.debug("==> addResourceIfReadable(" + aResourceName + ")");
-		}
-
-		URL fUrl = getFileLocation(aResourceName);
-		if (fUrl != null) {
-			if(LOG.isDebugEnabled()) {
-				LOG.debug("addResourceIfReadable(" + aResourceName + "): resource file is " + fUrl);
-			}
-			try {
-				addResource(fUrl);
-				ret = true;
-			} catch (Exception e) {
-				LOG.error("Unable to load the resource name [" + aResourceName + "]. Ignoring the resource:" + fUrl);
-				if (LOG.isDebugEnabled()) {
-					LOG.debug("Resource loading failed for " + fUrl, e);
-				}
-			}
-		} else {
-			if(LOG.isDebugEnabled()) {
-				LOG.debug("addResourceIfReadable(" + aResourceName + "): couldn't find resource file location");
-			}
-		}
-
-		if(LOG.isDebugEnabled()) {
-			LOG.debug("<== addResourceIfReadable(" + aResourceName + "), result=" + ret);
-		}
-		return ret;
-	}
-	
-	public Properties getProperties() {
-		return getProps();
-	}
-
-	private URL getFileLocation(String fileName) {
-		URL lurl = null;
-		if (!StringUtils.isEmpty(fileName)) {
-			lurl = RangerConfiguration.class.getClassLoader().getResource(fileName);
-
-			if (lurl == null ) {
-				lurl = RangerConfiguration.class.getClassLoader().getResource("/" + fileName);
-			}
-
-			if (lurl == null ) {
-				File f = new File(fileName);
-				if (f.exists()) {
-					try {
-						lurl=f.toURI().toURL();
-					} catch (MalformedURLException e) {
-						LOG.error("Unable to load the resource name [" + fileName + "]. Ignoring the resource:" + f.getPath());
-					}
-				} else {
-					if(LOG.isDebugEnabled()) {
-						LOG.debug("Conf file path " + fileName + " does not exists");
-					}
-				}
-			}
-		}
-		return lurl;
-	}
+    private static final Logger LOG = LoggerFactory.getLogger(RangerConfiguration.class);
+
+    protected RangerConfiguration() {
+        super(false);
+    }
+
+    public boolean addResourceIfReadable(String aResourceName) {
+        LOG.debug("==> addResourceIfReadable({})", aResourceName);
+
+        boolean ret  = false;
+        URL     fUrl = getFileLocation(aResourceName);
+
+        if (fUrl != null) {
+            LOG.debug("addResourceIfReadable({}): resource file is {}", aResourceName, fUrl);
+
+            try {
+                addResource(fUrl);
+
+                ret = true;
+            } catch (Exception e) {
+                LOG.error("Unable to load the resource name [{}]. Ignoring the resource:{}", aResourceName, fUrl);
+
+                LOG.debug("Resource loading failed for {}", fUrl, e);
+            }
+        } else {
+            LOG.debug("addResourceIfReadable({}): couldn't find resource file location", aResourceName);
+        }
+
+        LOG.debug("<== addResourceIfReadable({}), result={}", aResourceName, ret);
+
+        return ret;
+    }
+
+    public Properties getProperties() {
+        return getProps();
+    }
+
+    private URL getFileLocation(String fileName) {
```

Comment:
please see:

private URL getFileLocation(String fileName) {
    if (StringUtils.isEmpty(fileName)) {
        return null;
    }

    URL lurl = RangerConfiguration.class.getClassLoader().getResource(fileName);

    if (lurl == null) {
        lurl = RangerConfiguration.class.getClassLoader().getResource("/" + fileName);
    }

    if (lurl == null) {
        File file = new File(fileName);
        if (file.exists()) {
            try {
                return file.toURI().toURL();
            } catch (MalformedURLException e) {
                LOG.error("Unable to load the resource name [{}]. Ignoring the resource: {}", fileName, file.getPath());
            }
        } else {
            LOG.debug("Conf file path {} does not exist", fileName);
        }
    }

    return lurl;
}

Reply:
Current implementation is to keep a single return statement in the method. I suggest retaining this.